sematic/ui/src/pipelines/external_resource/ExternalResourceState.tsx

```diff
@@ -0,0 +1,28 @@
+import TimelineConnector from "@mui/lab/TimelineConnector/TimelineConnector";
+import TimelineContent from "@mui/lab/TimelineContent/TimelineContent";
+import TimelineDot from "@mui/lab/TimelineDot/TimelineDot";
+import TimelineItem from "@mui/lab/TimelineItem/TimelineItem";
+import TimelineSeparator from "@mui/lab/TimelineSeparator/TimelineSeparator";
+import { HistorySerializationType, TERMINATE_STATE } from "../../hooks/externalResourceHooks";
+import { format } from 'date-fns'
+import { useMemo } from "react";
+import Typography from "@mui/material/Typography/Typography";
+
+export default function ExternalResourceState({state}: {state: HistorySerializationType}) {
+    const {id, status_message, created_at} = state;
+    const timeString = useMemo(
+        () => format(
+            created_at instanceof Date ? created_at : new Date(created_at), 
+            'LLL\xa0d,\xa0yyyy\xa0h:mm:ss\xa0a'), [created_at]);
+
+    return <TimelineItem key={id}>
+        <TimelineSeparator>
+            <TimelineDot />
```

Comment:
Can the dots be colored depending on the state? For example:
- CREATED, DEACTIVATED: grey
- ACTIVATING, DEACTIVATING: blue
- ACTIVATED: green

Reply:
Yes, will add in the next iteration.